rfc/plugins.md

```diff
@@ -0,0 +1,235 @@
+# Plugins
+
+A plugin is an ember-cli addon that hooks into the ember-cli-deploy
+pipeline in order to add functionality to a deploy. Example addon
+functionality would be uploading assets to S3, writing the index.html to
+Redis, or notifying a Slack channel a deploy has completed.
+
+Plugins should focus on doing a specific task. Ember developers will
+compose and combine multiple plugins to fine tune their deployment
+process.
+
+### Identifying plugins
+
+Plugins are published to npm as ember-cli addons. They will also have
+the keyword ``ember-cli-deploy-plugin``.
+
+Any plugin that is installed will be loaded by default, similar to
+ember-cli addons.
+
+### Base plugin object
+
+Plugins will extend a base object provided by ember-cli-deploy. The API
```

Comment:
I like the idea of having a base Plugin that is extended.  Similar to extending an `ember-cli` Task.  But now you mention it @lukemelia, maybe it is weird to have a dependency on `ember-cli-deploy`.  But then again, it is a plugin for `ember-cli-deploy`, so maybe it isn't that weird.  This comment added zero value whatsoever ;)